src/app/core/duplicate-records/duplicate-records.service.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { TestBed } from '@angular/core/testing';
+import { DuplicateRecordService } from './duplicate-records.service';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service';
+import { DatabaseEntity, entityRegistry, EntityRegistry } from '../entity/database-entity.decorator';
+import { Database } from '../database/database'; 
+import { SessionService } from '../session/session-service/session.service';
+import { Entity } from '../entity/model/entity';
+import { DatabaseField } from '../entity/database-field.decorator';
+import { CoreModule } from '../core.module';
+import { ComponentRegistry } from 'app/dynamic-components';
+import { UpdateMetadata } from '../entity/model/update-metadata';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { FileService } from 'app/features/file/file.service';
+
+describe('DuplicateRecordsService', () => {
+  let service: DuplicateRecordService;
+  let entityMapperService: EntityMapperService;
+  let database: Database;
+
+  @DatabaseEntity("DuplicateTestEntity")
+  class DuplicateTestEntity extends Entity {
+  @DatabaseField() name: String;
+  @DatabaseField() boolProperty: boolean;
+  @DatabaseField() created: UpdateMetadata;
+  @DatabaseField() updated: UpdateMetadata;
+  @DatabaseField() inactive: boolean;
+  }
+
+  const schemaName = DuplicateTestEntity.ENTITY_TYPE;
+  
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [CoreModule],
+      providers: [
+        DuplicateRecordService,
+        Database,
+        EntityMapperService,
+        SessionService,
+        { provide: EntityRegistry, useValue: entityRegistry },
+        { provide: MatDialog, useValue: {} },
+        { provide: MatSnackBar, useValue: {} },
+        { provide: FileService, useValue: {} },
+        ComponentRegistry,]
+    });
+    service = TestBed.inject(DuplicateRecordService);
+    entityMapperService = TestBed.inject(EntityMapperService);
+    database = TestBed.inject(Database); 
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should transform data correctly', () => { 
+    const duplicateTest = new DuplicateTestEntity();
+    duplicateTest.name = "TestName"
+    duplicateTest.boolProperty = true;
+    
+    const originalData = [
+      {
+        record: duplicateTest
+      }
+    ];
```

Comment:
wouldn't things be easier if we implement the service to just work with an array of entities (`[duplicateTest]`) instead of this object?

Reply:
@sleidig, Updated it.